template.tpl

```diff
@@ -441,6 +502,23 @@ var initData = data.advancedMatchingParams && data.advancedMatchingParams.length
 initData.integration = 'gtm';
 initData.useDecimalCurrencyValues = true;
 
+var dataProcessingOptions = data.dataProcessingParams && data.dataProcessingParams.length ? makeTableMap(data.dataProcessingParams, 'name', 'value') : null;
+if (dataProcessingOptions && dataProcessingOptions.mode && dataProcessingOptions.mode.trim() !== '') {
+  var dpmArray = dataProcessingOptions.mode.split(',');
```

Comment:
I hate to say it, but I feel like this sort of business logic belongs in the pixel.js, not in the GTM template.  But I would happily defer to someone from the Signals team here.

Reply:
Yeah I agree.  It looks like for the advanced matching parameters, we are simply taking them in as is.  We should likely just do something like this on line 500:
```
var initData = {...makeTableMap(data.advancedMatchingParams || [], 'name', 'value'), ...makeTableMap(data.dataProcessingParams || [], 'name', 'value')};
```
To simplify this whole section.

Reply:
sg, moving it to pixel!

about using the `...makeTableMap(data.dataProcessingParams` google tag manager throws a warning that `object spread property` is only supported in ES9. So I'll keep the old way